src/L1/rollup/IScrollChain.sol

```diff
@@ -26,6 +26,22 @@ interface IScrollChain {
     /// @param withdrawRoot The merkle root on layer2 after this batch.
     event FinalizeBatch(uint256 indexed batchIndex, bytes32 indexed batchHash, bytes32 stateRoot, bytes32 withdrawRoot);
 
+    /// @notice Emitted when a batch is finalized by tee proof.
+    /// @param batchIndex The index of the batch.
+    event FinalizeBatchWithTEEProof(uint256 indexed batchIndex);
```

Comment:
I'd rather keep the semantics of `FinalizeBatch` the same as batch finalization (currently it's still used for ZKP verification, which is not considered final anymore).

I.e. 3 events:
- `VerifyBatchWithZkp`
- `VerifyBatchWithTee`
- `FinalizeBatch`

(These last two will coincide -- but this will make sure that dapp/node logic that relies on `FinalizeBatch` will continue to work as expected.)